src/lib.rs

```diff
@@ -399,14 +455,25 @@ impl LockFile {
     /// # }
     /// ```
     pub fn unlock(&mut self) -> Result<(), Error> {
+        debug_assert!(self.desc != sys::uninitialized_fd());
         if !self.locked {
             panic!("Attempted to unlock already locked lockfile");
         }
         self.locked = false;
         sys::unlock(self.desc)?;
-        sys::truncate(self.desc)?;
+        if self.truncate_on_close {
+            sys::truncate(self.desc)?;
+        }
         Ok(())
     }
+
+    /// It is recommended that you use the implementation [`<&mut LockFile as
+    /// Into<File>>::into`](struct.LockFile.html#impl-Into<File>) instead of
+    /// this low-level function.
+    pub unsafe fn raw(&self) -> FileDesc {
+        debug_assert!(self.desc != sys::uninitialized_fd());
+        self.desc
+    }
```

Comment:
Is this (method `.raw()`) really required? I think this goes out of the scope of a "retain PID" feature. And I'd also think it would leak internal API, which I'd rather keep private.